src/EditorFeatures/CSharpTest/UseLocalFunction/UseLocalFunctionTests.cs

```diff
@@ -1694,6 +1816,133 @@ public void Caller(T t)
             local(t);
         }
     }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        public async Task TestWithDefaultParameter1()
+        {
+           await TestInRegularAndScript1Async(
+@"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        MyDelegate [||]local = (s) => s;
+    }
+}",
+@"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        string local(string s = ""hello"") => s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        [WorkItem(24760, "https://github.com/dotnet/roslyn/issues/24760#issuecomment-364655480")]
+        public async Task TestWithDefaultParameter2()
+        {
+            await TestInRegularAndScript1Async(
+ @"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        MyDelegate [||]local = (string s) => s;
+    }
+}",
+ @"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        string local(string s = ""hello"") => s;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseLocalFunction)]
+        public async Task TestWithDefaultParameter3()
+        {
+            await TestInRegularAndScript1Async(
+ @"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        MyDelegate [||]local = delegate (string s) { return s; };
+    }
+}",
+ @"class C
+{
+    delegate string MyDelegate(string arg = ""hello"");
+
+    void M()
+    {
+        string local(string s = ""hello"")
+        { return s; }
```

Comment:
formatting seems non-great.  is this the same we do for lambdas?

Reply:
@CyrusNajmabadi yes... it's already reported here #23872

Reply:
do you want to take a stab at fixing that while you're here?  Basically, the logic i would have is:  if hte lambda body and parameter is on the same line, then use non-elastic trivia with the new local function that is created between the ```)``` and the body.

Reply:
i don't know how those things work

Reply:
Would you like to know more? :)

Reply:
sure

Reply:
i'm on gitter if you want to chat there :)